channeldb/payments.go

```diff
@@ -512,6 +513,28 @@ func serializeHop(w io.Writer, h *route.Hop) error {
 		return err
 	}
 
+	if err := binary.Write(w, byteOrder, h.LegacyPayload); err != nil {
+		return err
+	}
+
+	// For legacy payloads, we don't need to write any TLV records. so
+	// we'll write an empty byte slice.
+	if h.LegacyPayload {
+		if err := wire.WriteVarBytes(w, 0, []byte{}); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	var tlvBuf bytes.Buffer
+	if err := h.PackHopPayload(&tlvBuf); err != nil {
```

Comment:
Here we are serializing duplicate information to the database. My opinion remains that we are breaking the abstraction by doing this. I think we should either move to full tlv in the database for the new payload or just store the tlv custom fields.

Reply:
one option is to use an alternative stream that doesn't serialize the fundamental hop info, e.g. amt, cltv, and next hop. then we'd only be encoding "extra" data.